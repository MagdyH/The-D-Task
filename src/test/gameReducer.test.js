import configureStore from 'redux-mock-store';

const initialState = {}
const mockStore = configureStore();
const store = mockStore(initialState);
import GameReducer from '../model/game/reducers';
import GameAction from '../model/game/actions';
import ActionType from '../model/game/constants'


describe('Game Reducer', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
      });

      test("dataRequested", () => {
        const initialState = [{"match_id":"169254","country_id":"124","country_name":"Scotland","league_id":"423","league_name":"Premiership","match_date":"2019-01-23","match_status":"Finished","match_time":"20:45","match_hometeam_id":"6580","match_hometeam_name":"Celtic","match_hometeam_score":"4","match_awayteam_name":"St. Mirren","match_awayteam_id":"6598","match_awayteam_score":"0","match_hometeam_halftime_score":"2","match_awayteam_halftime_score":"0","match_hometeam_extra_score":"2","match_awayteam_extra_score":"0","match_hometeam_penalty_score":"","match_awayteam_penalty_score":"","match_hometeam_system":"4 - 2 - 3 - 1","match_awayteam_system":"4 - 1 - 4 - 1","match_live":"0","goalscorer":[{"time":"11","home_scorer":"Burke O.","score":"1 - 0","away_scorer":""},{"time":"18","home_scorer":"Sinclair S.","score":"2 - 0","away_scorer":""},{"time":"55","home_scorer":"Burke O.","score":"3 - 0","away_scorer":""}],"cards":[{"time":"58","home_fault":"Benkovic F.","card":"yellow card","away_fault":""}],"substitutions":{"home":[],"away":[{"time":"17","substitution":"Hodson L.  | Kellerman J."}]},"lineup":{"home":{"starting_lineups":[{"lineup_player":"Ajer K.","lineup_number":"35","lineup_position":"3"},{"lineup_player":"Bain S. (G)","lineup_number":"29","lineup_position":"1"},{"lineup_player":"Benkovic F.","lineup_number":"32","lineup_position":"4"},{"lineup_player":"Brown S. (C)","lineup_number":"8","lineup_position":"6"},{"lineup_player":"Burke O.","lineup_number":"25","lineup_position":"11"},{"lineup_player":"Christie R.","lineup_number":"17","lineup_position":"9"},{"lineup_player":"Forrest J.","lineup_number":"49","lineup_position":"8"},{"lineup_player":"Izaguirre E.","lineup_number":"3","lineup_position":"5"},{"lineup_player":"Lustig M.","lineup_number":"23","lineup_position":"2"},{"lineup_player":"McGregor C.","lineup_number":"42","lineup_position":"7"},{"lineup_player":"Sinclair S.","lineup_number":"11","lineup_position":"10"}],"substitutes":[{"lineup_player":"Bitton N.","lineup_number":"6","lineup_position":""},{"lineup_player":"Gordon C. (G)","lineup_number":"1","lineup_position":""},{"lineup_player":"Hayes J.","lineup_number":"15","lineup_position":""},{"lineup_player":"Henderson E.","lineup_number":"52","lineup_position":""},{"lineup_player":"Johnston M.","lineup_number":"73","lineup_position":""},{"lineup_player":"Simunovic J.","lineup_number":"5","lineup_position":""},{"lineup_player":"Weah T.","lineup_number":"30","lineup_position":""}],"coach":[{"lineup_player":"Rodgers B.","lineup_number":"-1","lineup_position":""}],"missing_players":[{"lineup_player":"Edouard O.","lineup_number":"-1","lineup_position":""},{"lineup_player":"Ntcham O.","lineup_number":"-1","lineup_position":""},{"lineup_player":"Ralston A.","lineup_number":"-1","lineup_position":""},{"lineup_player":"Rogic T.","lineup_number":"-1","lineup_position":""},{"lineup_player":"Tierney K.","lineup_number":"-1","lineup_position":""}]},"away":{"starting_lineups":[{"lineup_player":"Baird J.","lineup_number":"15","lineup_position":"3"},{"lineup_player":"Cooke C.","lineup_number":"20","lineup_position":"11"},{"lineup_player":"Erhahon E.","lineup_number":"39","lineup_position":"5"},{"lineup_player":"Flynn R.","lineup_number":"8","lineup_position":"10"},{"lineup_player":"Hladky V. (G)","lineup_number":"1","lineup_position":"1"},{"lineup_player":"Hodson L.","lineup_number":"21","lineup_position":"6"},{"lineup_player":"Lyons B.","lineup_number":"14","lineup_position":"8"},{"lineup_player":"McGinn P.","lineup_number":"2","lineup_position":"2"},{"lineup_player":"McGinn S. (C)","lineup_number":"4","lineup_position":"9"},{"lineup_player":"Popescu M.","lineup_number":"16","lineup_position":"4"},{"lineup_player":"Tansey G.","lineup_number":"22","lineup_position":"7"}],"substitutes":[{"lineup_player":"Ferdinand A.","lineup_number":"35","lineup_position":""},{"lineup_player":"Holmes J. (G)","lineup_number":"26","lineup_position":""},{"lineup_player":"Jackson S.","lineup_number":"11","lineup_position":""},{"lineup_player":"Kellerman J.","lineup_number":"17","lineup_position":""},{"lineup_player":"McAllister K.","lineup_number":"23","lineup_position":""}],"coach":[{"lineup_player":"Kearney O.","lineup_number":"-1","lineup_position":""}],"missing_players":[{"lineup_player":"MacKenzie G.","lineup_number":"-1","lineup_position":""},{"lineup_player":"Magennis K.","lineup_number":"-1","lineup_position":""},{"lineup_player":"Mullen D.","lineup_number":"-1","lineup_position":""}]}},"statistics":[{"type":"Ball Possession","home":"72%","away":"28%"},{"type":"Goal Attempts","home":"19","away":"1"},{"type":"Shots on Goal","home":"9","away":"1"},{"type":"Shots off Goal","home":"7","away":"0"},{"type":"Blocked Shots","home":"3","away":"0"},{"type":"Free Kicks","home":"13","away":"16"},{"type":"Corner Kicks","home":"8","away":"2"},{"type":"Goalkeeper Saves","home":"1","away":"5"},{"type":"Fouls","home":"14","away":"9"},{"type":"Yellow Cards","home":"1","away":"1"}]}];

        const expectedActions = [{
          type: ActionType.GET_ALL_GAMES,
          games: [{"match_id":"169254","country_id":"124","country_name":"Scotland","league_id":"423","league_name":"Premiership","match_date":"2019-01-23","match_status":"Finished","match_time":"20:45","match_hometeam_id":"6580","match_hometeam_name":"Celtic","match_hometeam_score":"4","match_awayteam_name":"St. Mirren","match_awayteam_id":"6598","match_awayteam_score":"0","match_hometeam_halftime_score":"2","match_awayteam_halftime_score":"0","match_hometeam_extra_score":"2","match_awayteam_extra_score":"0","match_hometeam_penalty_score":"","match_awayteam_penalty_score":"","match_hometeam_system":"4 - 2 - 3 - 1","match_awayteam_system":"4 - 1 - 4 - 1","match_live":"0","goalscorer":[{"time":"11","home_scorer":"Burke O.","score":"1 - 0","away_scorer":""},{"time":"18","home_scorer":"Sinclair S.","score":"2 - 0","away_scorer":""},{"time":"55","home_scorer":"Burke O.","score":"3 - 0","away_scorer":""}],"cards":[{"time":"58","home_fault":"Benkovic F.","card":"yellow card","away_fault":""}],"substitutions":{"home":[],"away":[{"time":"17","substitution":"Hodson L.  | Kellerman J."}]},"lineup":{"home":{"starting_lineups":[{"lineup_player":"Ajer K.","lineup_number":"35","lineup_position":"3"},{"lineup_player":"Bain S. (G)","lineup_number":"29","lineup_position":"1"},{"lineup_player":"Benkovic F.","lineup_number":"32","lineup_position":"4"},{"lineup_player":"Brown S. (C)","lineup_number":"8","lineup_position":"6"},{"lineup_player":"Burke O.","lineup_number":"25","lineup_position":"11"},{"lineup_player":"Christie R.","lineup_number":"17","lineup_position":"9"},{"lineup_player":"Forrest J.","lineup_number":"49","lineup_position":"8"},{"lineup_player":"Izaguirre E.","lineup_number":"3","lineup_position":"5"},{"lineup_player":"Lustig M.","lineup_number":"23","lineup_position":"2"},{"lineup_player":"McGregor C.","lineup_number":"42","lineup_position":"7"},{"lineup_player":"Sinclair S.","lineup_number":"11","lineup_position":"10"}],"substitutes":[{"lineup_player":"Bitton N.","lineup_number":"6","lineup_position":""},{"lineup_player":"Gordon C. (G)","lineup_number":"1","lineup_position":""},{"lineup_player":"Hayes J.","lineup_number":"15","lineup_position":""},{"lineup_player":"Henderson E.","lineup_number":"52","lineup_position":""},{"lineup_player":"Johnston M.","lineup_number":"73","lineup_position":""},{"lineup_player":"Simunovic J.","lineup_number":"5","lineup_position":""},{"lineup_player":"Weah T.","lineup_number":"30","lineup_position":""}],"coach":[{"lineup_player":"Rodgers B.","lineup_number":"-1","lineup_position":""}],"missing_players":[{"lineup_player":"Edouard O.","lineup_number":"-1","lineup_position":""},{"lineup_player":"Ntcham O.","lineup_number":"-1","lineup_position":""},{"lineup_player":"Ralston A.","lineup_number":"-1","lineup_position":""},{"lineup_player":"Rogic T.","lineup_number":"-1","lineup_position":""},{"lineup_player":"Tierney K.","lineup_number":"-1","lineup_position":""}]},"away":{"starting_lineups":[{"lineup_player":"Baird J.","lineup_number":"15","lineup_position":"3"},{"lineup_player":"Cooke C.","lineup_number":"20","lineup_position":"11"},{"lineup_player":"Erhahon E.","lineup_number":"39","lineup_position":"5"},{"lineup_player":"Flynn R.","lineup_number":"8","lineup_position":"10"},{"lineup_player":"Hladky V. (G)","lineup_number":"1","lineup_position":"1"},{"lineup_player":"Hodson L.","lineup_number":"21","lineup_position":"6"},{"lineup_player":"Lyons B.","lineup_number":"14","lineup_position":"8"},{"lineup_player":"McGinn P.","lineup_number":"2","lineup_position":"2"},{"lineup_player":"McGinn S. (C)","lineup_number":"4","lineup_position":"9"},{"lineup_player":"Popescu M.","lineup_number":"16","lineup_position":"4"},{"lineup_player":"Tansey G.","lineup_number":"22","lineup_position":"7"}],"substitutes":[{"lineup_player":"Ferdinand A.","lineup_number":"35","lineup_position":""},{"lineup_player":"Holmes J. (G)","lineup_number":"26","lineup_position":""},{"lineup_player":"Jackson S.","lineup_number":"11","lineup_position":""},{"lineup_player":"Kellerman J.","lineup_number":"17","lineup_position":""},{"lineup_player":"McAllister K.","lineup_number":"23","lineup_position":""}],"coach":[{"lineup_player":"Kearney O.","lineup_number":"-1","lineup_position":""}],"missing_players":[{"lineup_player":"MacKenzie G.","lineup_number":"-1","lineup_position":""},{"lineup_player":"Magennis K.","lineup_number":"-1","lineup_position":""},{"lineup_player":"Mullen D.","lineup_number":"-1","lineup_position":""}]}},"statistics":[{"type":"Ball Possession","home":"72%","away":"28%"},{"type":"Goal Attempts","home":"19","away":"1"},{"type":"Shots on Goal","home":"9","away":"1"},{"type":"Shots off Goal","home":"7","away":"0"},{"type":"Blocked Shots","home":"3","away":"0"},{"type":"Free Kicks","home":"13","away":"16"},{"type":"Corner Kicks","home":"8","away":"2"},{"type":"Goalkeeper Saves","home":"1","away":"5"},{"type":"Fouls","home":"14","away":"9"},{"type":"Yellow Cards","home":"1","away":"1"}]}]
          
        }];
        store.dispatch(GameAction.getAllGames(initialState));
        expect(store.getActions()).toEqual(expectedActions);
});

      test('is working', () => {
        const actionSearch = { type: ActionType.GET_ALL_GAMES };
        const actionGetClubs = { type: ActionType.SEARCH_GAME };
    
        const initialStateGetClubs = { games: [],storedGames:[] };
        const initialStateSearch = { games: undefined,storedGames:[] };
    
        expect(GameReducer(undefined, actionSearch)).toEqual(initialStateSearch);
        expect(GameReducer(undefined, actionGetClubs)).toEqual(initialStateGetClubs);
    
        expect(GameReducer(undefined, actionSearch)).toMatchSnapshot();
        expect(GameReducer(undefined, actionGetClubs)).toMatchSnapshot()
      });
});