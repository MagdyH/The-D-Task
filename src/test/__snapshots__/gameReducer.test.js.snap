// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Reducer is working 1`] = `
Object {
  "games": undefined,
  "storedGames": Array [],
}
`;

exports[`Game Reducer is working 2`] = `
Object {
  "games": Array [],
  "storedGames": Array [],
}
`;
